pyo3-build-config/src/impl_.rs

```diff
@@ -733,39 +828,74 @@ pub(crate) fn cross_compile_env_vars() -> CrossCompileEnvVars {
 ///   `PYO3_CROSS_LIB_DIR`.
 ///
 /// See the [PyO3 User Guide](https://pyo3.rs/) for more info on cross-compiling.
+///
+/// TODO: Deprecate in v0.17 favor of `cross_compiling_from_to()`.
```

Comment:
Can't we deprecate this right now (as the alternative API is added) or would adding a deprecation be a breaking change?

(I also think this should get a change log entry as it modifies the public API of `pyo3-build-config` even most users will not interact with it directly.)

Reply:
> Can't we deprecate this right now (as the alternative API is added) or would adding a deprecation be a breaking change?

It depends on whether this commit gets into the next point release or it is first released in v0.17.
I can replace `TODO:` with `#[deprecated(since="0.17.0", note="please use cross_compile_from_to() instead")]`.

I'm adding a ChangeLog entry in the next push.

Reply:
I've added a deprecation attribute and a ChangeLog entry.